ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -446,6 +458,32 @@ internal sealed class ArchiWebHandler : IDisposable {
 			return await UrlGetToHtmlDocumentWithSession(SteamCommunityURL, request).ConfigureAwait(false);
 		}
 
+		internal async Task<HashSet<ulong>> GetDigitalGiftCards() {
+			const string requestGifts = "/gifts";
+			HtmlDocument response = await UrlGetToHtmlDocumentWithSession(SteamStoreURL, requestGifts).ConfigureAwait(false);
+			if (response == null) {
+				return null;
+			}
+
+			HtmlNodeCollection nodes = response.DocumentNode.SelectNodes("//div[@class='pending_gift']/div[starts-with(@id,'pending_gift_')]/div[@class='pending_giftcard_leftcol']/../@id");
+			HashSet<ulong> results = new HashSet<ulong>();
+			foreach (string gidText in nodes.Select(node => node.GetAttributeValue("id", null)?.Substring(13))) {
```

Comment:
XPath guarantees you that ID will start with `pending_gift_` so this is correct as it is, but I'd rather have just `GetAttributeValue()` here, then below null check as it is, and then another check if length is greater than 13, and then substring call. This way you won't call substring if ID has just `pending_gift_`, and knowing Valve and Steam, I'd not be shocked if it decided to do such stuff at some point. You could take idea from AWH `GetApiKeyState()`.